data/store/mongo/mongo_test.go

```diff
@@ -237,35 +237,42 @@ func DataSetDatumAsInterface(dataSetDatum data.Datum) interface{} {
 }
 
 var _ = Describe("Mongo", Label("mongodb", "slow", "integration"), func() {
-	var logger *logTest.Logger
-	var config *storeStructuredMongo.Config
-	var store *dataStoreMongo.Store
 	var repository dataStore.DataRepository
 	var summaryRepository dataStore.SummaryRepository
 	var alertsRepository alerts.Repository
+	var logger = logTest.NewLogger()
+	var initOnce, cleanupOnce sync.Once
+	var store *dataStoreMongo.Store
 
 	BeforeEach(func() {
-		logger = logTest.NewLogger()
-		config = storeStructuredMongoTest.NewConfig()
+		initOnce.Do(func() {
+			config := storeStructuredMongoTest.NewConfig()
+			testStore, err := dataStoreMongo.NewStore(config)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(testStore).ToNot(BeNil())
+			Expect(testStore.EnsureIndexes()).To(Succeed())
+			store = testStore
+		})
 	})
 
 	AfterEach(func() {
-		if store != nil {
-			_ = store.Terminate(context.Background())
-		}
+		cleanupOnce.Do(func() {
+			storeStructuredMongoTest.DeferredCleanup(func() {
+				err := store.Terminate(context.Background())
+				Expect(err).To(Succeed())
+			})
+		})
 	})
 
 	Context("New", func() {
 		It("returns an error if unsuccessful", func() {
-			var err error
-			store, err = dataStoreMongo.NewStore(nil)
+			store, err := dataStoreMongo.NewStore(nil)
```

Comment:
[nit] This is a little confusion with `store` matching the same name as the more "global" `store`. Consider using `testStore` as you did above. Same for the test below.

Reply:
Or, you could pull these two tests out of the BeforeEach/AfterEach Describe. (You'd need to create a new Context with for most everything and put these outside of that.)

Reply:
Will do.

Reply:
To be clear, this change will appear in #777 not here.